ipa/prover.go

```diff
@@ -127,14 +127,21 @@ func CreateIPAProof(transcript *common.Transcript, ic *IPAConfig, commitment ban
 }
 
 // Write serializes the IPA proof to the given writer.
-func (ip *IPAProof) Write(w io.Writer) {
+func (ip *IPAProof) Write(w io.Writer) error {
 	for _, el := range ip.L {
-		binary.Write(w, binary.BigEndian, el.Bytes())
+		if err := binary.Write(w, binary.BigEndian, el.Bytes()); err != nil {
```

Comment:
Another thing found by the fuzzer. We were ignoring `binary.Write` potentially returning an error, thus continuing with the logic when we shouldn't. `binary.Write` can fail if the underlying `Writer` breaks when trying to be written.

So here now I check for the error, which forces `IPAProof.Write` to also return an error. It's just the right thing to do.